validation-test/stdlib/SetAlgebra.swift

```diff
@@ -0,0 +1,181 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+extension MinimalEquatableValue : Hashable {
+  public var hashValue: Int {
+    return value
+  }
+}
+
+public struct DefaultedSetAlgebra<Element : Hashable> : SetAlgebra {
+  var store = Set<Element>()
+  public init(){}
+  public init(_ elements: [Element]) {
+    for el in elements {
+      store.insert(el)
+    }
+  }
+
+  public func contains(_ member: Element) -> Bool {
+    return store.contains(member)
+  }
+
+  public func union(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
```

Comment:
Can you add an initializer `DefaultedSetAlgebra.init(_store: Set<Element>)`?  The underscore will prevent it from conflicting with anything in the library.

Then you can write these forwarding functions more easily, in just one line:

```
return DefaultedSetAlgebra(_store: store.union(other.store))
```